logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/UpdateCallStatusById.kt

```diff
@@ -0,0 +1,33 @@
+package com.wire.kalium.logic.data.call
+
+import com.wire.kalium.logic.data.id.toConversationId
+import com.wire.kalium.logic.feature.call.Call
+import com.wire.kalium.logic.feature.call.CallStatus
+import com.wire.kalium.logic.feature.call.calls
+import kotlinx.coroutines.flow.update
+
+//TODO create unit test for this one in a different PR
+class UpdateCallStatusById {
```

Comment:
Maybe we could rename this to something like `CallStateManager` as it is only handling the call state

Reply:
I don't really like to use some general names like `Manager`, `Handler` etc...It breaks the single responsibility  principle, I want to be more specific  : Manager could be creating, deleting or updating.
Also by using `UpdateCallStatusById` as a name, you don't have to open the file and see the what it does.

Reply:
I agree with @ohassine on avoiding generic names like Manager and Handler.

But when you have a instance of `UpdateCallStatusById`, how can you name it?

```kotlin
val update = UpdateCallStatusById()
update.updateById(id) { /** **/ }
```

Seems weird. 


When we name classes, we don't name them the same as a function, we name the classes and objects for what they are.

It is **not** an `Update`. It is an `Updater`. It performs updates:

```kotlin
val callUpdater = CallUpdater()
callUpdater.updateById(id) { /** **/ }
```


A silly comparison:

```kotlin
interface Cat {
    fun meow()
}
```

Instead of:
```kotlin
interface Meow {
    fun meow()
}
```

Reply:
The more I look at this, the more I want to put this logic inside `CallRepository`.

Because to be fair, this is what it is.
Everywhere else we are using Repositories for this.

Just because this is stored in memory instead of DB, SharedPreferences or API, doesn't mean it can't be dealt with in a Repository.

Bonus: we have plenty of tested Repositories already, so we know how to do it. And we can share for all platforms.

Reply:
Totally agree, Thanks 👌